src/webrtc/remoteCallFeed.ts

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2023 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { logger } from "../logger";
+import { CallEvent, CallState, MatrixCall } from "./call";
+import { SDPStreamMetadataObject, SDPStreamMetadataPurpose } from "./callEventTypes";
+import { CallFeed, CallFeedEvent, ICallFeedOpts } from "./callFeed";
+import { RemoteCallTrack } from "./remoteCallTrack";
+
+export interface RemoteCallFeedOpts extends ICallFeedOpts {
+    streamId: string;
+    metadata?: SDPStreamMetadataObject;
+    call: MatrixCall;
+
+    /**
+     * @deprecated addTransceiver() should be used instead
+     */
+    stream?: MediaStream;
+}
+
+export class RemoteCallFeed extends CallFeed {
+    private _connected = false;
+    private _metadata?: SDPStreamMetadataObject;
+
+    protected _tracks: RemoteCallTrack[] = [];
+    protected call: MatrixCall;
+    protected _stream: MediaStream;
+
+    public readonly streamId: string;
+    public readonly isLocal = false;
+    public readonly isRemote = true;
+
+    public constructor(opts: RemoteCallFeedOpts) {
+        super(opts);
+
+        if (!opts.metadata && opts.call.opponentSupportsSDPStreamMetadata()) {
+            throw new Error(
+                "Cannot create RemoteCallFeed without metadata if the opponents supports sending sdp_stream_metadata",
+            );
+        }
+
+        this.streamId = opts.streamId;
+        this.call = opts.call;
+        this.metadata = opts.metadata;
+
+        this._stream = opts.stream || new window.MediaStream();
```

Comment:
I think probably just `MediaStream` as there's no `window` in node (assuming we'd ever do webrtc in node, but may as well follow best practice).

Reply:
This is analogous to `new window.RTCPeerConnection()` in `MatrixCall::createPeerConnection`. When running jest, `MediaStream` was undefined and the only way to get it working was this along with the mock in `installWebRTCMocks()`

Reply:
oh, if we use `window` elsewhere then fair enough.